### ISSUE TITLE
Internationalisation

### ISSUE THREAD

**Issue by markusr** [grafana/grafana#448]
Hi,

how can we add a translation (language file) of the GUI? There is not a lot in English but our customer prefers German language.

If someone explains how to translate I will make the translation to German. Maybe other users will contribute other languages.

Thank you 
 MarkusR

**Comment by thomas-huang** [comment]
+1

**Comment by Kuermel** [comment]
+1

**Comment by nucleusv** [comment]
+1

**Comment by toni-moreno** [comment]
+1
I will be able to make spanish and catalan translations.

**Comment by markusr** [comment]
I have done some research. 

I suggest using [angular-translate](https://github.com/angular-translate/angular-translate) and to do the translation itself in transifex or something similar.

**Comment by The-Nik** [comment]
I would also like to have a support of German :-)

**Comment by petersonbr** [comment]
+1

**Comment by derkan** [comment]
+1

**Comment by bergquist** [comment]
I've worked in a lot of projects with internationalisation and it tends to slow down the development speed immensely. But I do understand the need for consuming information in your native language. 

Perhaps we can have documentation in different languages rather than the product? It's 99% about graphs and numbers anyway :)  //cc @torkelo @nopzor1200

**Comment by n3bojs4** [comment]
+1

**Comment by wnasich** [comment]
+1

**Comment by samosvat** [comment]
+1

**Comment by MichaelSavin** [comment]
+1

**Comment by marisradu** [comment]
+1

**Comment by peterhuerlimann** [comment]
+1

**Comment by kinhunt** [comment]
+1

**Comment by wbelguidoum** [comment]
+1 .. i can provide french and arabic translation

**Comment by henriquemaciel** [comment]
+1

Portuguese - Brazil

**Comment by davyjones** [comment]
+1

`ja_JP.utf8`

**Comment by Virtex7** [comment]
+1
and another issue within this: at the moment grafana does not support the german "umlauts" as in äöü. This is an issue because if i use them in the alias field of a graphs legend the whole graph does not work as expected and throws an "internal server error".
now i have changed the umlauts to their representation but it is no nice solution.

**Comment by jisaitua** [comment]
+1

**Comment by zphj1987** [comment]
+1

**Comment by cznewt** [comment]
+2

**Comment by PiR43** [comment]
+1

**Comment by rojosewe** [comment]
+1

**Comment by t-pfaff** [comment]
+1

**Comment by David-BEM** [comment]
Another (kind of) show stopper ist that the CSV-export needs internationalization: It's frustrating but in german speaking countries the "," is used to seperate decimals where internationally it's the ".". The issue is that Excel/LibreOffice with german language settings will import the numbers as text. :-/
Also: yes, german language for the GUI would be awesome !

**Comment by messier51a** [comment]
Any advances with internationalization?. I'm about to start working on a translation script but before doing so I wanted to check if anyone here already has something.

**Comment by JasonKeirstead** [comment]
+1. Also wondering if you are looking for assistance in doing this. 

This would be a showstopper for us; without i18n I can not use Grafana in a project we are looking at.

**Comment by dadosch** [comment]
+1

**Comment by bgruenefeld** [comment]
+1

**Comment by jacktang** [comment]
+1

**Comment by RafaelLeonhardt** [comment]
+1

**Comment by hzhaop** [comment]
+1

**Comment by pronichev** [comment]
+1
I will make the translation to Russian.

**Comment by gavin-hu** [comment]
+1

**Comment by leechdraw** [comment]
+1
ready to help with Russian traslation

**Comment by YuehShan-Chung** [comment]
+1

**Comment by pomarc** [comment]
+1
this lack of localization really prevents us from using grafana in one of our projects.

**Comment by StephenOTT** [comment]
👍 +2

**Comment by cnsunfocus** [comment]
+1

**Comment by zhaoyim** [comment]
+1

**Comment by liusong1111** [comment]
+1

**Comment by leafonsword** [comment]
+1

**Comment by macmeaker** [comment]
+1

**Comment by wei-hai** [comment]
+1

**Comment by rudoc** [comment]
+1

**Comment by iShift** [comment]
+1

**Comment by Trubochkin** [comment]
+1

**Comment by leohihimax** [comment]
+1

**Comment by vpmayer** [comment]
+1

**Comment by bcw104** [comment]
+1

**Comment by sylforever98** [comment]
+ 翻译中文

**Comment by jordimas** [comment]
+1

**Comment by tangfang** [comment]
+1

**Comment by ferrang** [comment]
+1

**Comment by changjiasheng** [comment]
+1

**Comment by byakuinss** [comment]
+1

**Comment by Nayar** [comment]
I have somewhat managed to translate Grafana into french using `angular-translate`. It's working like 90%. 

![spectacle l12531](https://user-images.githubusercontent.com/454815/30541932-629904be-9c8e-11e7-8d6f-1531f8c29524.png)

I can't translate the calendar right now:

![spectacle d12531](https://user-images.githubusercontent.com/454815/30541971-923479c4-9c8e-11e7-906a-ad2e5026ad52.png)


I have a file in which I write the translations for each language:

```
app.config(['$translateProvider', function ($translateProvider) {
      $translateProvider.translations('fr', {
        'resetmail': 'Mot de passe oublié?',
        'User' : "Utilisateur",
        "Password" : "Mot de passe",
        "emailorusername" : "mail ou identifiant",
        "Back to dashboard" : "Retour au tableau de bord",
```

I pipe the English texts in the HTMLs like this:

```
<span class="gf-form-label width-7">{{'User' | translate}}</span>
<span class="gf-form-label width-7">{{'Password' | translate }}</span>
```

```
<select class="gf-form-input" ng-model="ctrl.prefs.timezone" ng-options="f.value as (f.text | translate) for f in ctrl.timezones">
```

I'll clean my mess a bit and post my codes later on.

**Comment by trusnak** [comment]
+1

**Comment by svzain** [comment]
@Nayar I was able to change almost everything but the dropdown menu and the calendar. Did you find the solution? 
![grafana](https://user-images.githubusercontent.com/32128769/32044669-2716ade8-ba79-11e7-851e-67d7da51092d.PNG)

**Comment by Hyruu** [comment]
@Nayar What's the status on your code cleaning ? Would definitely be interested in the solution you exposed, and I think in a first time it will cover the needs of most people having posted here.

**Comment by rlcomte** [comment]
+1
We like the angular-translate approach. Would you still recommend to have different dashboard content for different languages? We have the requirement to have both Grafana UX and the dashboard itself in a specific language (Dutch, German, French).

**Comment by Nayar** [comment]
I'm putting the barebones in this branch: https://github.com/Nayar/grafana/commits/i18n

I already translated the login form. 

![screenshot_20171212_162104](https://user-images.githubusercontent.com/454815/33884264-7593135e-df58-11e7-953d-6e469ece8de1.png)

**Comment by Naesstrom** [comment]
+1

**Comment by dfisek** [comment]
+1

**Comment by sureshjanarthan** [comment]
+1

**Comment by IAlexandr** [comment]
+1 russian

**Comment by carlosbaia** [comment]
+1

**Comment by pedrommone** [comment]
Internationalization will slow down the development only if the main team take care of it.

Many others project let the comunity translate with tools like [crowdin](https://crowdin.com).

**Comment by kaven9** [comment]
+1 for russian translation. e-mail me when ready

**Comment by TayZ25** [comment]
+1 Japanese

**Comment by sundoforce** [comment]
+1 Korean

**Comment by lunamay** [comment]
+1

**Comment by swucim** [comment]
+1 for Chinese language!

**Comment by odin533** [comment]
+1 for Chinese language package

**Comment by wangmuqi** [comment]
+1  for Chinese language package

**Comment by ychost** [comment]
+1 for Chinese language package

**Comment by HughWen** [comment]
+1 for Chinese language package

**Comment by Archieeeeee** [comment]
+1 for Chinese language package

**Comment by leinad87** [comment]
+1

**Comment by hpham04** [comment]
+1

**Comment by chris922** [comment]
Would be great if you will not spam with "+1" comments. If you like this you can give a 'thumbs up' to the first post. Otherwise it would be hard to start a discussion about the feature request here if there are thousands of "+1" comments in between. Thanks!

**Comment by yunayena** [comment]
Hi Torkelo! I have a request, today is there able to internationalization? I read a lot of post but i don´t have clearly if it is possible. Thank you!

**Comment by ivankostov** [comment]
+1 bulgarian

**Comment by mohsenasm** [comment]
+1 persian

**Comment by evandoloiola** [comment]
+1 portuguese - BR

**Comment by datianjiao** [comment]
+中文

**Comment by liangmuren** [comment]
+1 for Chinese language package

**Comment by XQDD** [comment]
+1

**Comment by a459295360** [comment]
+1

**Comment by Ancitik** [comment]
+1 for french support 👍

**Comment by nekijak** [comment]
+1 for Japanese language package

**Comment by tammytankian** [comment]
+1 Portuguese - BR

**Comment by mingzhou** [comment]
+1 for Chinese language package

**Comment by xikenmai** [comment]
+1 for Simplified Chinese

**Comment by zhil** [comment]
+1

**Comment by vincent-zurczak** [comment]
Hi,

It seems Grafana uses AngularJS. [Angular Translate](https://github.com/angular-translate/angular-translate) is a solution to bring internationalization in AngularJS applications. I already used it and it works well. I also developed [Angular Translate Quality](https://github.com/linagora/angular-translate-quality), a NodeJS plug-in you can add in your build process to verify JS sources, HTML templates and translation files.

Internationalized dashboards is a frequent requirement from clients. For public administrations and organizations, it may even be mandatory (that's the law). Assuming one submits a pull request to add i18n in Grafana, and provided it is compliant with your development guidelines, would you accept it? It also implies you would rely on the community to maintain translation files. Is it something you would be ready to accept?

BTW, do you plan to keep AngularJS? If you want to migrate to another web framework, internationalization would have to be reset and doing the job right now would make no sense.

**Comment by torkelo** [comment]
We are migrating React, large parts of Grafana is already in react

**Comment by vincent-zurczak** [comment]
OK, thanks. I had limited my checks to the first message and the `package.json` file.
Question: migrating implies upgrading what exists. Did you consider working on internationalization while migrating?

**Comment by Adilbek1988** [comment]
+1

**Comment by vincenzomanzoni** [comment]
+1

**Comment by sloomy** [comment]
+1 for French

**Comment by gokr** [comment]
Just a tip - in Canoe (getcanoe.io) we decided to use POEditor.com and let people sign up and help out with translations. It works great and we have more than 100 people helping out translating. POEditor is free for open source projects. I can't say it slows down dev much, if at all.

**Comment by Exotelis** [comment]
@sundoforce 
This could be a timing issue. The translate provider might run before the title has been returned. Try to find a way to rerun the translation loop. The 'tsx' file seems to be the problem here, because I don't think that there is a way to load the angular service into this file.

**Comment by ntting** [comment]
+1 for Simplified Chinese

**Comment by bassmaster187** [comment]
+1

**Comment by hankchang47106** [comment]
+1 for Traditional Chinese

**Comment by lucasmogari** [comment]
+1

**Comment by Rene-Auberger** [comment]
+1

**Comment by greeneyz** [comment]
@sundoforce or anyone who has implemented the translation successfully in a running project:
Can you please help in the implementation step by step? I want to know exactly how to create the new translation file and where and how to update it so that it can reflect the project.

**Comment by zoesn89** [comment]
+1

**Comment by monitorelx** [comment]
> > I have somewhat managed to translate Grafana into french using `angular-translate`. It's working like 90%.

That's great! Could you please share your files so we can translate Grafana into other languages? Maybe that way we are able to find solutions to the problems you are facing. Thanks in advance!

**Comment by ntting** [comment]
+1

**Comment by erduoxiangshang** [comment]
+1

**Comment by zhoumin8023** [comment]
+1
+1
+1

**Comment by erduoxiangshang** [comment]
+1

**Comment by bergquist** [comment]
Please stop spamming this issue. Vote on the issue instead. 

We are currently rewriting the frontend of Grafana from angular into react so it's very unlike that we would consider this right now.

**Comment by wilmermantilla** [comment]
> > I have somewhat managed to translate Grafana into french using `angular-translate`. It's working like 90%.
> > ![spectacle l12531](https://user-images.githubusercontent.com/454815/30541932-629904be-9c8e-11e7-8d6f-1531f8c29524.png)
> > I can't translate the calendar right now:
> > ![spectacle d12531](https://user-images.githubusercontent.com/454815/30541971-923479c4-9c8e-11e7-906a-ad2e5026ad52.png)
> > I have a file in which I write the translations for each language:
> > ```
> > app.config(['$translateProvider', function ($translateProvider) {
> >       $translateProvider.translations('fr', {
> >         'resetmail': 'Mot de passe oublié?',
> >         'User' : "Utilisateur",
> >         "Password" : "Mot de passe",
> >         "emailorusername" : "mail ou identifiant",
> >         "Back to dashboard" : "Retour au tableau de bord",
> > ```
> > I pipe the English texts in the HTMLs like this:
> > ```
> > <span class="gf-form-label width-7">{{'User' | translate}}</span>
> > <span class="gf-form-label width-7">{{'Password' | translate }}</span>
> > ```
> > Even works for loops
> > ```
> > <select class="gf-form-input" ng-model="ctrl.prefs.timezone" ng-options="f.value as (f.text | translate) for f in ctrl.timezones">
> > ```
> > I'll clean my mess a bit then post the codes later
> 
> ======================================
> 
> I got a lot of help.
> thank you .
> and mostly succeed
> 
> but
> I did not succeed.
> pageHeader (tsx file)
> 
> Did you fix this?
> I need your help.
> 
> ` return <h1 className="page-header__title">{title}</h1>;`
> 
> ->
> `return <h1 className="page-header__title">{title | translate}</h1>;`
> 
> https://stackoverflow.com/questions/53476757/angular-translate-in-grafana-headerpage-modify-tsx-file

hello cordial, it will be that you can give me a hand how to translate. grafana to Spanish, the datepicker and the timepicker.

**Comment by tweaker** [comment]
++++

**Comment by pcbl** [comment]
+1

**Comment by juannvalencia** [comment]
1+ for translate to Spanish

**Comment by killdaclick** [comment]
I have got translated *.html files from "grafana/public" directory but I found out they are not enough. Most of readable text are inside "grafana/build/*.js" in compacted form.
Is it possible to include my translations into grafana?

**Comment by gretamosa** [comment]
It would be a complete Grafana front-end refactor but it is an [idea](https://www.freecodecamp.org/news/setting-up-internationalization-in-react-from-start-to-finish-6cb94a7af725/). Where can we vote an issue?

**Comment by MFullStack** [comment]
+1 for translate to chinese

**Comment by MFullStack** [comment]
> > > I have somewhat managed to translate Grafana into french using `angular-translate`. It's working like 90%.
> > > ![spectacle l12531](https://user-images.githubusercontent.com/454815/30541932-629904be-9c8e-11e7-8d6f-1531f8c29524.png)
> > > I can't translate the calendar right now:
> > > ![spectacle d12531](https://user-images.githubusercontent.com/454815/30541971-923479c4-9c8e-11e7-906a-ad2e5026ad52.png)
> > > I have a file in which I write the translations for each language:
> > > ```
> > > app.config(['$translateProvider', function ($translateProvider) {
> > >       $translateProvider.translations('fr', {
> > >         'resetmail': 'Mot de passe oublié?',
> > >         'User' : "Utilisateur",
> > >         "Password" : "Mot de passe",
> > >         "emailorusername" : "mail ou identifiant",
> > >         "Back to dashboard" : "Retour au tableau de bord",
> > > ```
> > > 
> > > 
> > > I pipe the English texts in the HTMLs like this:
> > > ```
> > > <span class="gf-form-label width-7">{{'User' | translate}}</span>
> > > <span class="gf-form-label width-7">{{'Password' | translate }}</span>
> > > ```
> > > 
> > > 
> > > Even works for loops
> > > ```
> > > <select class="gf-form-input" ng-model="ctrl.prefs.timezone" ng-options="f.value as (f.text | translate) for f in ctrl.timezones">
> > > ```
> > > 
> > > 
> > > I'll clean my mess a bit then post the codes later
> > 
> > 
> > ======================================
> > I got a lot of help.
> > thank you .
> > and mostly succeed
> > but
> > I did not succeed.
> > pageHeader (tsx file)
> > Did you fix this?
> > I need your help.
> > ` return <h1 className="page-header__title">{title}</h1>;`
> > ->
> > `return <h1 className="page-header__title">{title | translate}</h1>;`
> > https://stackoverflow.com/questions/53476757/angular-translate-in-grafana-headerpage-modify-tsx-file
> 
> hello cordial, it will be that you can give me a hand how to translate. grafana to Spanish, the datepicker and the timepicker.

I has understand angular-translate works in angular. but grafana has many TSX file of react, how angular-translate works in it?

**Comment by patricksferraz** [comment]
+1
Portuguese Brazil

**Comment by 190948804** [comment]
+1

**Comment by killdaclick** [comment]
I have got translation for 100% o texts found in html files for Polish language. Ready to implement but need support??

**Comment by noger-one** [comment]
+1
I can help with Slovak translation sk_SK

**Comment by mqgmaster** [comment]
@bergquist, so, the i18n is totally discarded for the future Grafana with React? From my experience, the i18n is very easy if considered from the project start.

**Comment by wilfrid-dubreuil** [comment]
+1

**Comment by oyilmaztekin** [comment]
+1

**Comment by dsipahioglu** [comment]
+1

**Comment by grayfullbuster841998** [comment]
+1 for translate to vietnamese

**Comment by stevenvachon** [comment]
I think that the architecture for this could be done as an additional part of the e2e changes we're doing. We could officially provide only English language files per project (core and plugins) while other languages could be community maintained plugins. @torkelo

**Comment by onevetka** [comment]
+1

**Comment by qgzhao** [comment]
+1

**Comment by sebargarcia** [comment]
+1 for Spanish translation

**Comment by zibens** [comment]
+1 for Latvian translation

**Comment by dongbum** [comment]
+1

I can help with Korean translation

**Comment by kestouf** [comment]
+1
If I need help with the French translation, I can help.

**Comment by dennisfinning** [comment]
+1 for Spanish (South America) translation, and believe it or not, English UK (using different date formats, and potentially a few terms are different)

Has anyone looked at adding support for https://localizejs.com into Grafana? That way, it'd everyone could manage their own translations and the Grafana core team wouldn't need to spend extra cycles continually having to maintain different locales.

**Comment by nguyenngodinh** [comment]
+1

**Comment by lanliu99** [comment]
+1

**Comment by hderler1** [comment]
+ 1

**Comment by socketpair** [comment]
+1 for Russian.

You can use https://momentjs.com for date\time formatting. For other things there is gettext. Ask me for help if you want.

**Comment by socketpair** [comment]
> I think that the architecture for this could be done as an additional part of the e2e changes we're doing. We could officially provide only English language files per project (core and plugins) while other languages could be community maintained plugins. @torkelo

Yep, Please change the architecture. The actual translations could be done by simple pull requests. made by contributors. Gettext gives no errors on untranslated parts, so you should not fix all the translations on any code change. Also, gettext is perfect on plural forms.

**Comment by bryanasdev000** [comment]
+1 from Brazil

**Comment by JNVoirol** [comment]
+1

**Comment by aerosadegh** [comment]
+1
I can help to provide Persian Translation. :-)

**Comment by aerosadegh** [comment]
> +1
> I can help to provide Persian Translation. :-)

After it, I would like to help you to add Jalali calendar! 💯

**Comment by zoesn89** [comment]
@torkelo is there any update after all this time? Would really appreciate it! :)

**Comment by pppranik** [comment]
+1 for Ukrainian.

**Comment by sdellachiesa** [comment]
+1 Italian & German

**Comment by andrew-kratik** [comment]
+1 for Ukrainian and Russian

**Comment by ChinaShrimp** [comment]
+1

**Comment by socketpair** [comment]
https://github.com/date-fns/date-fns

is better than momentjs

**Comment by orangesunny** [comment]
As hundreds of users of Grafana are willing to do the translation of this wonderful open source software, we in [Weblate](https://github.com/WeblateOrg/weblate) are saying that we would be happy to help. Localization doesn’t have to slow down the development or increase the workload of the developers. It can be done easily, automatically, by the community and the libre way.

I see that there are already some `Qt Linguist translation files`, language `JSONs`, and `HTML` language files in Grafana repo. 😃  Once the i18n of the code is done 🤞 , we are prepared to provide the libre l10n platform.

What do you say Grafana team?

**Comment by mymtak** [comment]
+1 japanese

**Comment by fdepierre** [comment]
+1 I can help for french. I you want to delegate the translation project to a sub group of contributors, I can lead it for 2 years.

**Comment by matpandino** [comment]
+1 for portuguese

**Comment by daikunjp** [comment]
+1

**Comment by jehutywong** [comment]
+1

**Comment by mfandre** [comment]
+1

**Comment by keamspring** [comment]
+1 for Chinese language!

**Comment by agenteDserrano** [comment]
+1 Spanish

**Comment by fdepierre** [comment]
I was stepping back after making real projects with Grafana and creating dashboards.
Only the viewer access role would need to be translated: this type of users work in context with their language and switching to Grafana in EN could be difficult.
For users who make configuration changes, translation has no sense and It would be too difficult to maintain. These user can understand they are advanced users and work in EN.

**Comment by noger-one** [comment]
> I was stepping back after making real projects with Grafana and creating dashboards.
> Only the viewer access role would need to be translated: this type of users work in context with their language and switching to Grafana in EN could be difficult.
> For users who make configuration changes, translation has no sense and It would be too difficult to maintain. These user can understand they are advanced users and work in EN.

I agree with you. Translation is important only for viewer's

**Comment by orangesunny** [comment]
> I was stepping back after making real projects with Grafana and creating dashboards.
> Only the viewer access role would need to be translated: this type of users work in context with their language and switching to Grafana in EN could be difficult.
> For users who make configuration changes, translation has no sense and It would be too difficult to maintain. These user can understand they are advanced users and work in EN.

SW localization is my daily bread and the opinion you shared is usually gone after starting the localization. Not a lot of project owners keep that it long.

Yes, the end-user interface is the most important and definitely the best place to start with l10n. But no one ever got hurt by localized admin console or even documentation. Communities love to help with this! Ask Godot Engine project or check https://docs.weblate.org/ in many languages. Yes, people usually speak English these days, but their proficiency differs. And some of them, even technically skillful, don’t speak English at all. Higher comfort means happier users of the SW. And using the native language brings comfort.

I agree that end-user/viewer UI is a good place to `start` with Grafana localization. 😉

**Comment by leogaggl** [comment]
Just to add another one vote for the importance of multi-language support here is my +1 to make this a higher priority.

Also would be happy to contribute to German translation once available. 

As for i18n for viewers only I don't think that is a long-term option. As an initial priority maybe. But probably not in the long-term.

**Comment by MauroDiamantino** [comment]
+1 I could help with the translation to spanish.

**Comment by strippo** [comment]
+1 I could help with the Italian translation!

**Comment by sergio-chumacero** [comment]
+1 could help with Spanish translation

**Comment by tbotalla** [comment]
+1 can help with spanish

**Comment by igroman787** [comment]
+1

**Comment by guille50** [comment]
+1

**Comment by MarcoA12000** [comment]
+1

**Comment by jvitor83** [comment]
+1

**Comment by bruce0205** [comment]
+1

**Comment by CircleMan** [comment]
+1 I can help for japanese.

**Comment by sbollain-mlt** [comment]
+1  I could help with the translation to spanish.

**Comment by lopezsalexander** [comment]
+1 I could help with the translation to spanish.

**Comment by huajiejin** [comment]
+1 I could help with the translation to Chinese.

**Comment by izzetkalic** [comment]
+1 I could help with the translation to Turkish.

**Comment by joaocrramos** [comment]
+1

**Comment by tec-lab** [comment]
+1

**Comment by DmitrySidorow** [comment]
+1 for russian. can help with translation

**Comment by ajoneba** [comment]
+1 Japanese Translation

**Comment by agenteDserrano** [comment]
Spanish+

**Comment by Bruno-Guimaraes** [comment]
+1 for Portuguese translation.

**Comment by al-indigo** [comment]
+1

**Comment by Huseriato** [comment]
+1

**Comment by Dud3z** [comment]
+1 ...
so this issue is like 6 yrs old. please add at least the possibility to change values and time and date formatting...

**Comment by LauJangit** [comment]
+1
Multi language is quite essential for a GUI based software, make more language avalible can make grafana more popular in worldwide.

**Comment by DmitrySidorow** [comment]
We can easily translate grafana with simple scripts. Just translated timepicker for russian

![Screenshot_1](https://user-images.githubusercontent.com/43705385/103354491-be2f2580-4abc-11eb-91da-1edb8c2aabab.png)

**Comment by berlinsaint** [comment]
> We can easily translate grafana with simple scripts. Just translated timepicker for russian
> 
> ![Screenshot_1](https://user-images.githubusercontent.com/43705385/103354491-be2f2580-4abc-11eb-91da-1edb8c2aabab.png)

cool! can you share your expierence?

**Comment by DmitrySidorow** [comment]
> > We can easily translate grafana with simple scripts. Just translated timepicker for russian
> > ![Screenshot_1](https://user-images.githubusercontent.com/43705385/103354491-be2f2580-4abc-11eb-91da-1edb8c2aabab.png)
> 
> cool! can you share your expierence?

https://github.com/DmitrySidorow/GrafanaTranslator
Until an official translation is made, this is the easiest way to translate important parts. I hope someone will help me finish the translation

**Comment by strippo** [comment]
> > > We can easily translate grafana with simple scripts. Just translated timepicker for russian
> > > ![Screenshot_1](https://user-images.githubusercontent.com/43705385/103354491-be2f2580-4abc-11eb-91da-1edb8c2aabab.png)
> > 
> > 
> > cool! can you share your expierence?
> 
> https://github.com/DmitrySidorow/GrafanaTranslator
> Until an official translation is made, this is the easiest way to translate important parts. I hope someone will help me finish the translation

I've done a pull request for italian language, thx!

**Comment by awaili** [comment]
it is mandatory to add the localisation to the next version.
the user can select the language based on their interest.

**Comment by mertbakir** [comment]
+1 We need this feature too, especially for **"Units"** and **"Time Picker"**.

**Comment by SebastianBugna** [comment]
+1 feature super needed ! Grafana in Spanish please !

**Comment by SiNaPsEr0x** [comment]
can you add italian language ?

**Comment by EvdSvyat** [comment]
+1 Ukrainian

**Comment by ProteusCortex** [comment]
+1 Greek

**Comment by clamstor** [comment]
+1

**Comment by Incrediblez7** [comment]
+1

**Comment by a67793581** [comment]
+1 Chinese

**Comment by neuvill** [comment]
+1 can help with french

**Comment by krzd7** [comment]
A very important feature indeed.

Why not adding this in a way the community can handle language packs? I mean, Grafana should only support english by default, but having a way to upload your own language pack. It should not slow down development speed.

**Comment by liuyuran** [comment]
+1

**Comment by fernandomullerjr** [comment]
+1
Portuguese

**Comment by mohsenasm** [comment]
There was already a pull request for internationalization (and more) https://github.com/grafana/grafana/pull/3529. But it seems the maintainers didn't merge that with no explanation. So sad.

How should be a proper pull request for this feature?
There are many people in this thread that want to contribute.

**Comment by fellipedemoraes** [comment]
> +1
> 
> Portuguese - Brazil

> +1

**Comment by krzd7** [comment]
@torkelo, is there an update on this feature request?
I'm sure a lot of users would contribute to translation packages if we get a proper/official way of doing this.

**Comment by alexeysi** [comment]
+1 Russian... The UI is getting used As a Production dashboard where differently educated peoples are working, and it is common that usually they are speaking local language...Ever it may be OK if they will not be able to admin dashboards, but from end user point of view all UI should be in local language... From my prospective it is the must and show stopper feature!!!

**Comment by 1574242600** [comment]
+1

**Comment by Antohnio123** [comment]
> We can easily translate grafana with simple scripts. Just translated timepicker for russian
> 
> Will share soon.
> 
> ![Screenshot_1](https://user-images.githubusercontent.com/43705385/103354491-be2f2580-4abc-11eb-91da-1edb8c2aabab.png)

What happened?  The link is down. Is Translator for Russian available elsewhere?
We really need an opportunity to translate to Russian.

**Comment by izzetkalic** [comment]
> > We can easily translate grafana with simple scripts. Just translated timepicker for russian
> > Will share soon.
> > ![Screenshot_1](https://user-images.githubusercontent.com/43705385/103354491-be2f2580-4abc-11eb-91da-1edb8c2aabab.png)
> 
> What happened? The link is down. Is Translator for Russian available elsewhere?
> We really need an opportunity to translate to Russian.
> I'm ready to do this if the DmitrySidorow's code is unavailable.

Here my fork is available.

https://github.com/izzetkalic/GrafanaTranslator

**Comment by Antohnio123** [comment]
> > > We can easily translate grafana with simple scripts. Just translated timepicker for russian
> > > Will share soon.
> > > ![Screenshot_1](https://user-images.githubusercontent.com/43705385/103354491-be2f2580-4abc-11eb-91da-1edb8c2aabab.png)
> > 
> > 
> > What happened? The link is down. Is Translator for Russian available elsewhere?
> > We really need an opportunity to translate to Russian.
> > I'm ready to do this if the DmitrySidorow's code is unavailable.
> 
> Here my fork is available.
> 
> https://github.com/izzetkalic/GrafanaTranslator

Nice, but is there a Windows version of this script?

**Comment by Antohnio123** [comment]
> Here my fork is available.
> 
> https://github.com/izzetkalic/GrafanaTranslator
Should i rune it inside a grafana container which is currently running?
I get "Permission denied" error on every file, that should be changed. What do i do wrong?

**Comment by Antohnio123** [comment]
> Here my fork is available.
> 
> https://github.com/izzetkalic/GrafanaTranslator
![image](https://user-images.githubusercontent.com/59763511/124281068-073a6200-db52-11eb-9c14-f5935fd8f6da.png)

Nice, it worked out well. But main menu is still in English.
Will the contributors develop a project and translate Main menu first?

**Comment by Antohnio123** [comment]
Added a russian_1 branch with extended translation in file "langs/Russian_windows_bash.sh".  
https://github.com/Antohnio123/GrafanaTranslator/tree/russian_1/langs

I used it on Windows OS with Git Bash, started as Administrator. But you can easily return Linux folder in the beginning. 
![изображение](https://user-images.githubusercontent.com/59763511/124582249-87b2d880-de5a-11eb-8258-a8eb9f2df08c.png)
![изображение](https://user-images.githubusercontent.com/59763511/124582294-939e9a80-de5a-11eb-9c7a-f2a9bc6036bd.png)
Translating side menu has turned to be the most difficult thing, cause the files of it seem to be outside of grafana folder.

**Comment by zhenjie** [comment]
+1

**Comment by crazy-pizza** [comment]
+1

**Comment by jkmnt** [comment]
POC: https://github.com/jkmnt/grafana_i18n

**Comment by fengzhao** [comment]
+1 chinese

**Comment by gzwxh** [comment]
+1

**Comment by kajla** [comment]
+1

**Comment by fnoopv** [comment]
+1

**Comment by Mimmi511** [comment]
+1 
can't really believe that Grafana doesn't support any i18n

**Comment by krzd7** [comment]
this has been the most wanted feature for years now

please, consider checking if the POC posted by jkmnt meets all the requirements: https://github.com/jkmnt/grafana_i18n

**Comment by pdrivom** [comment]
+1
Portuguese - Portugal

**Comment by Bluefish67** [comment]
+1
German - Germany

**Comment by patogit** [comment]
In case i18n ever happens (maybe based on that proof of concept?), then I recommend using [TranslateWiki.net](https://translatewiki.net/) for localization. It's a translation platform for free/libre software that integrates with GitHub. An important thing to remember is to create documentation that helps translators understand what they're translating, by using the `qqq` language code.

**Comment by yarons** [comment]
@patogit No TM and some other pretty essential functions, better go with [Weblate](https://www.weblate.org) in that case.

**Comment by patogit** [comment]
@yarons Have you used the TranslateWiki.net interface to translate anything? It offers suggestions based on similar phrases that are already translated -- not sure if that qualifies as a translation memory or not. The benefit of TW.net is that there's a large community of translators already active on the platform. We put software on TW.net, and within a week it was already translated into more than ten languages completely by the volunteer translator community.

**Comment by sangjiexun** [comment]
> +1
> 
> 我将翻译为简体中文。

实现了么大佬

**Comment by Senlan0528** [comment]
> > +1
> > 我将翻译为简体中文。
> 
> 实现了么大佬

蹲

**Comment by gitty8** [comment]
+1 Germany

**Comment by KennyChan94P** [comment]
im planning to use Grafana in 4 languages (English, French, Spanish and Chinese). please advice on any workarounds that can achieve this. knowing that our web application has been developed in English only.

**Comment by sangjiexun** [comment]
感谢您的来信，我会马上回复您

**Comment by janckerchen** [comment]
+1 chinese

**Comment by kajla** [comment]
+1 Hungarian

**Comment by samuelting** [comment]
+1 for chinese

**Comment by suess-ecbm** [comment]
Looks like there are preparations for i18n going on! Great job!
https://github.com/grafana/grafana/pull/41827

**Comment by sangjiexun** [comment]
感谢您的来信，我会马上回复您

**Comment by Tmadben** [comment]
Hi everyone, I will be pleased to work on French translation.

**Comment by sangjiexun** [comment]
感谢您的来信，我会马上回复您

**Comment by xsa-dev** [comment]
+

**Comment by samuelting** [comment]
Hello everyone, I would be happy to offer help in Chinese Traditional translation.

**Comment by xfengfeng** [comment]
+1 Chinese

**Comment by lvvgang** [comment]
+1 Chinese

**Comment by shopkueche** [comment]
+1 German

**Comment by StephanGR** [comment]
+1 French

**Comment by sangjiexun** [comment]
感谢您的来信，我会马上回复您

**Comment by HanSolo72** [comment]
+1 German. Would also provide translation work.

**Comment by gsbecerrag** [comment]
+1 Spanish

**Comment by sangjiexun** [comment]
感谢您的来信，我会马上回复您

**Comment by jackbi** [comment]
+1 简体中文

**Comment by sparkdevo** [comment]
+1 简体中文

**Comment by Jia-SH** [comment]
+1 chinese

**Comment by sicoyle** [comment]
+1

**Comment by xxx123xxxx** [comment]
+1

**Comment by nicojahn** [comment]
+1

**Comment by ftoledo** [comment]
+1

**Comment by Nichsen** [comment]
+1

**Comment by Nemecsek** [comment]
+1 German

**Comment by achim-70** [comment]
+1 German

**Comment by scysys** [comment]
https://github.com/grafana/grafana/blob/main/contribute/internationalization.md

> To update phrases in any translated language, edit the phrase in Crowdin

Dont find Grafana on Crowdin......

**Comment by RubenGamezTorrijos** [comment]
It will be a pleasure to be able to join as a contributor for the translation of Grafana in Spanish🤗🤗🤗🤗

**Comment by btreeworld** [comment]
+1 

I can help with translate to Korean.

**Comment by joshhunt** [comment]
@scysys Apologies for the confusion - we're not yet ready to accept outside contributions for translations.

For our first lot of languages, Grafana will take care of the translations. We would like to eventually accept community translations, but it's quite an undertaking for us to be able to support that, so we don't have anything more to say about that yet.

**Comment by feng-yifan** [comment]
@joshhunt May I ask for a list of languages which was planned to be supported? And when will it be released?

**Comment by morfeurj** [comment]
Hey guys.
I can help with the translation to Portuguese/Brazil.
Is anyone working on this translation now?

**Comment by samuelting** [comment]
@joshhunt, May I know whether there is any plan / roadmap to support internationalisation in near terms?

**Comment by Radik24** [comment]
+1 Czech/Slovak

I can help with translation.

**Comment by joshhunt** [comment]
We will share proper details when we enable it by default, but we have translations for 'viewers' of Grafana for French, Spanish, German, and Chinese (Simplified).

These are currently in the main branch (if you run grafana from source) behind the `internationalization` [feature toggle](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#feature_toggles), and we expect to have this enabled by default soon in an upcoming minor release. @feng-yifan @samuelting

**Comment by Parinat** [comment]
@joshhunt  we also have a problem with thai language. 
Can you include it in this release.

**Comment by kajla** [comment]
+1 Hungarian

We can help with translation.

**Comment by feng-yifan** [comment]
> We will share proper details when we enable it by default, but we have translations for 'viewers' of Grafana for French, Spanish, German, and Chinese (Simplified).
> 
> These are currently in the main branch (if you run grafana from source) behind the `internationalization` [feature toggle](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#feature_toggles), and we expect to have this enabled by default soon in an upcoming minor release. @feng-yifan @samuelting

Thanks for replay, hopefully the translations for 'editor' will also be supported asap :)

**Comment by samuelting** [comment]
> We will share proper details when we enable it by default, but we have translations for 'viewers' of Grafana for French, Spanish, German, and Chinese (Simplified).
> 
> These are currently in the main branch (if you run grafana from source) behind the `internationalization` [feature toggle](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#feature_toggles), and we expect to have this enabled by default soon in an upcoming minor release. @feng-yifan @samuelting

Do you minding supporting Chinese Traditional in the coming releases?

**Comment by joshhunt** [comment]
In Grafana 9.3 we will release our first pass at internationalisation for a few key workflows, so we feel this is a good time to close this issue. 

For any issues with the quality of translations, please raise a new issue to this Github so we can triage and investigate appropriately. 

As mentioned before, we are not currently accepting outside contributions for translators. 

If you have an opinion on what part of Grafana we should translate next, please share your thoughts in this discussion https://github.com/grafana/grafana/discussions/59227

**Comment by gomiunik** [comment]
Perhaps you can get help translating from community with https://weblate.org. I've been cooperating on some other open source project to aid in translation and I'd be happy to help here as well
